File: src/PlantBasedPizza.Api/modules/recipes/PlantBasedPizza.Recipes.Core/IntegrationEvents/RecipeCreatedEventV1.cs
using System.Text.Json.Serialization;
using PlantBasedPizza.Events;

namespace PlantBasedPizza.Recipes.Core.IntegrationEvents;

public class RecipeCreatedEventV1 : IntegrationEvent
{
    public override string EventName => "recipe.recipeCreated";
    public override string EventVersion => "v1";
    public override Uri Source => new Uri("https://recipes.plantbasedpizza");
    
    [JsonPropertyName("RecipeIdentifier")]
    public string RecipeIdentifier { get; init; } = "";
}

File: src/PlantBasedPizza.Recipes.Java/src/core/src/main/java/com/recipe/core/RecipeCreatedEventV1.java
package com.recipe.core;

public class RecipeCreatedEventV1 {
    private long recipeId;

    public RecipeCreatedEventV1(){
        this.recipeId = -1;
    }

    public RecipeCreatedEventV1(long recipeId){
        this.recipeId = recipeId;
    }

    public long getRecipeId() {
        return recipeId;
    }

    public void setRecipeId(long recipeId) {
        this.recipeId = recipeId;
    }
}
